src/visualisation/panel.cc

```diff
@@ -230,97 +241,127 @@ void VisualisationPanel::refreshVisualisation() {
 void VisualisationPanel::initLayout() {
   initOptionsPanel();
 
-  QWidget* root_widget = new QWidget(this);
-  this->setCentralWidget(root_widget);
-
-  splitter_ = new QSplitter(Qt::Horizontal);
-  splitter_->addWidget(minimap_);
-  splitter_->addWidget(visualisation_);
-  splitter_->setStretchFactor(1, 5);
-
-  layout_ = new QHBoxLayout;
-  layout_->setContentsMargins(0, 0, 0, 0);
-  layout_->addWidget(splitter_, 8);
-  layout_->addLayout(options_layout_, 0);
-  root_widget->setLayout(layout_);
+  setCentralWidget(visualisation_root_);
+  setDockNestingEnabled(true);
+
+  node_tree_dock_ = new QDockWidget;
```

Comment:
QMainWindow::addDockWidget(w) calls QLayout::addChildWidget(w).
If w doesn't have a parent, QLayout::addChildWidget(w) calls w->setParent() with a widget that owns the layout as a parameter. So this is correct.